### DISCUSSION TITLE
Question: how access context / req data in Processors and apply global scopes?

### DISCUSSION THREAD

**Discussion by renanwilliam** [kurierjs/kurier#331] [Q&A] [answered]
First of all, thanks for sharing the lib; it's incredible! 

I'm setting my first project using it with Knex default processor, and I have two situations challenging me:
- I need to inject user information to apply filters based on account. But my authentication is not with JWT, it is using cookies (ory.sh). I understand that the [built-in method](https://kurier.readthedocs.io/en/latest/#authentication-and-authorization) is only for JWT, and it won't work for me. I understand that I can rewrite the [jsonApiKoa middleware](https://github.com/kurierjs/kurier/blob/main/src/middlewares/json-api-koa.ts) as a local middleware and adapt to inject using my cookies/session approach. Rewriting this file in my project makes sense, or is it not made to be changed/extended, and the library is not designed for that?
- I need to apply a scope in my [CustomProcessor extended from KnexProcessor](https://kurier.readthedocs.io/en/latest/#the-knexprocessor-class) based on user tenant ID (user only can see your data - so a filter needs to be applied in all queries). Is there an example of something like that? The only way to get user information is with `this.appInstance.user` ? Access requests headers or other information is possible inside a CustomProcessor?

Thanks in advance, and sorry if I created duplicated questions.

**Comment by joelalejandro** [comment, marked as answer]
Hey @renanwilliam!

- You could implement your own `jsonApiKoa` middleware in your own project. Unfortunately the current version doesn't expose `http-utils`, which you'd need to create your own processor. We can export those.
- Scoping data is possible through the UserManagementAddon, using the IfUser helpers as described in the docs: https://kurier.readthedocs.io/en/latest/#authentication-and-authorization. Basically what it does is to apply conditional checks over `this.appInstance.user`.
